Move scripts from dev/ into ivadomed/scripts (#289)
neuropoly/ivadomed: docs/source/scripts.rst

..  scripts:

Utility scripts
===============

> Visualize data augmentation transformations

Reviewer: not sure how they appear on the left TOC, but it is fairly small, so i would keep titles short, e.g. "Visualize transformations" or "Visualize augmentation transfos"
but this is a very minor detail...

Reviewer: update: after looking at the generated built doc, it looks fine the way it is 👍

***********************************************
Data augmentation is a key part of the Deep Learning training scheme. This script aims at facilitating the fine-tuning of data augmentation parameters. To do so, this script provides a step-by-step visualization of the transformations that are applied on data.

The ``scripts/visualize_transforms.py`` script applies a series of transformations (defined in a configuration file ``-c``) to ``-n`` 2D slices randomly extracted from an input image (``-i``), and save as png the resulting sample after each transform.

For example::

    python scripts/visualize_transforms.py -i t2s.nii.gz -n 1 -c config.json -r t2s_seg.nii.gz

> Provides a visualisation of a series of three transformation on a randomly selected slice:

Reviewer: ```suggestion
Provides a visualization of a series of three transformation on a randomly selected slice:
```

.. image:: ../../images/transforms_im.png

>     :width: 200px

Reviewer: this needs to be much bigger. let's try this:
```suggestion
    :width: 600px
```

    :align: center

>     :height: 100px

Reviewer: why specifying both width and height? you want to keep the ratio
```suggestion
```

And on a binary mask::

    python scripts/visualize_transforms.py -i t2s_gmseg.nii.gz -n 1 -c config.json -r t2s_seg.nii.gz

Gives:

.. image:: ../../images/transforms_gt.png

>     :width: 200px

Reviewer: ```suggestion
    :width: 600px
```

    :align: center

>     :height: 100px

Reviewer: ```suggestion
```

> Convert a model from PyTorch to ONNX format

Reviewer: same here about length, how about "Convert to ONNX"
but the more i think about it, the more i like the "verbose" approach. Not sure how it will appear on the TOC. we could ask RTD to build the doc for this branch. i'll do it now...

***********************************************
The integration of Deep Learning models into the clinical routine requires cpu optimized models. To export the PyTorch models to `ONNX <https://github.com/onnx/onnx>`_ format and to run the inference using `ONNX Runtime <https://github.com/microsoft/onnxruntime>`_ is a time and memory efficient way to answer this need.

``scripts/convert_to_onnx.py`` converts a model from PyTorch to ONNX format, with information of whether it is a 2D or 3D model (``-d``)::

    python scripts/convert_to_onnx.py -m path/to/model.pt -d 3

> Model's hyper-parameters optimisation

Reviewer: i more often see hyperparameter as one word-- and also to keep US convention (z)
```suggestion
Hyperparameter optimization
```

***********************************************

> Model's hyper-parameters optimisation is tedious and time-consuming. ``scripts/automate_training.py`` allows to automatize this optimisation on multiple GPUs. The scripts runs trainings, on the same training and validation datasets, by combinating a given set of parameters and set of values for each of these parameters. Results are collected for each combination and reported into a dataframe to allow their comparison. The script efficiently allocates each training to one of the available GPUs. ::

Reviewer: ```suggestion
Hyperparameter optimization of models is tedious and time-consuming. ``scripts/automate_training.py`` automatizes this optimization across multiple GPUs. The scripts runs trainings, on the same training and validation datasets, by combining a given set of parameters and set of values for each of these parameters. Results are collected for each combination and reported into a dataframe to allow their comparison. The script efficiently allocates each training to one of the available GPUs. ::
```

    python scripts/automate_training.py -c path/to/config.json

.. TODO: add example of DF

> Compare models' performance at inference time

Reviewer: ```suggestion
Compare models performance at inference time
```
